cli/tests/activate.bats

```diff
@@ -2836,6 +2836,47 @@ EOF
   assert_output --partial "3.2.2"
 }
 
+@test "in-place activate works with bash 3" {
+  if ! /bin/bash --version | grep -q "GNU bash, version 3"; then
+    skip "No bash 3 at /bin/bash"
+  fi
```

Comment:
(non-blocking, observation) I'm cautious of conditionals skips because we won't know if the behaviour changes, but it seems unlikely that Darwin is going to change any time soon and I can't see a way to get that version of Bash into `flox-cli-tests` in parallel. Confirmed that it's not currently skipped on either of our Darwin builders:

- https://github.com/flox/flox/actions/runs/14412081464/job/40422545033#step:5:262
- https://github.com/flox/flox/actions/runs/14412081464/job/40422543822#step:5:355

Reply:
Might be better to fail the test if it's on macOS and /bin/bash isn't version 3

I think testing against /bin/bash is honestly better than testing against something Nix provided, because no one is going to be using Nix provided bash 3 😂